libbeat/publisher/pipeline/controller.go

```diff
@@ -136,12 +137,13 @@ func (c *outputController) Set(outGrp outputs.Group) {
 
 	// restart consumer (potentially blocked by retryer)
 	c.consumer.sigContinue()
+	c.consumer.sigUnWait()
```

Comment:
`sigUnWait` is used by the retryer, no? Why add it here?

Reply:
I found that this was needed to make the `TestReload` test pass. Without it the retryer never un-waited the consumer by itself and so events stopped being consumed and sent to the output.

That said, I agree that this is something the retryer should automatically call internally rather than "leak" this responsibility to the controller. I'll look into why it's not doing that.

Reply:
Yeah, this sounds like a potential bug to me.

Reply:
Addressed in 6a6680c.